examples/large_models/inferentia2/llama2/continuous_batching/inf2_handler.py

```diff
@@ -0,0 +1,332 @@
+        )
+        model_checkpoint_path = f"{model_dir}/{model_checkpoint_dir}"
+        model_path = f'{model_dir}/{ctx.model_yaml_config["handler"]["model_path"]}'
+
+        if not os.path.exists(model_checkpoint_path):
+            # Load and save the CPU model
+            model_cpu = AutoModelForCausalLM.from_pretrained(
+                model_path, low_cpu_mem_usage=True
+            )
+            save_pretrained_split(model_cpu, model_checkpoint_path)
+            # Load and save tokenizer for the model
+            tokenizer = AutoTokenizer.from_pretrained(
+                model_path, return_tensors="pt", padding_side="left"
+            )
+            tokenizer.save_pretrained(model_checkpoint_path)
+
+        os.environ["NEURONX_CACHE"] = "on"
+        os.environ["NEURON_COMPILE_CACHE_URL"] = f"{model_dir}/neuron_cache"
+        os.environ["NEURON_CC_FLAGS"] = "-O1 --model-type=transformer"
+
+        self.max_length = int(
+            ctx.model_yaml_config.get("handler", {}).get("max_length", self.max_length)
+        )
+        self.max_new_tokens = int(
+            ctx.model_yaml_config.get("handler", {}).get(
+                "max_new_tokens", self.max_new_tokens
+            )
+        )
+        self.batch_size = int(
+            ctx.model_yaml_config.get("handler", {}).get("batch_size", self.batch_size)
+        )
+
+        # settings for model compilation and loading
+        amp = ctx.model_yaml_config.get("handler", {}).get("amp", "fp32")
+        tp_degree = ctx.model_yaml_config.get("handler", {}).get("tp_degree", 6)
+        context_length_estimate = ctx.model_yaml_config.get("handler", {}).get(
+            "context_length_estimate", self.max_length
+        )
+
+        # allocate "tp_degree" number of neuron cores to the worker process
+        os.environ["NEURON_RT_NUM_CORES"] = str(tp_degree)
+        try:
+            num_neuron_cores_available = (
+                torch_neuronx.xla_impl.data_parallel.device_count()
+            )
+            assert num_neuron_cores_available >= int(tp_degree)
+        except (RuntimeError, AssertionError) as error:
+            logger.error(
+                "Required number of neuron cores for tp_degree "
+                + str(tp_degree)
+                + " are not available: "
+                + str(error)
+            )
+
+            raise error
+
+        self.tokenizer = LlamaTokenizer.from_pretrained(
+            model_checkpoint_path, return_tensors="pt", padding_side="left"
+        )
+        self.tokenizer.pad_token = self.tokenizer.eos_token
+        self.tokenizer.pad_token_id = self.tokenizer.eos_token_id
+        self.tokenizer.padding_side = "left"
+        self.model = LlamaForSampling.from_pretrained(
+            model_checkpoint_path,
+            batch_size=self.batch_size,
+            amp=amp,
+            tp_degree=tp_degree,
+            n_positions=self.max_length,
+            context_length_estimate=context_length_estimate,
+        )
```

Comment:
I think continuous batching won't work without configuring `neuron_config`.

Here is a typical example on how to configure it:
```python
# Load the weights from the cached or downloaded files.
from transformers_neuronx.config import NeuronConfig, ContinuousBatchingConfig

continuous_batching_config = ContinuousBatchingConfig(batch_size_for_shared_caches=scheduler_config.max_num_seqs)
neuron_config = NeuronConfig(continuous_batching=continuous_batching_config)
model.load_weights(model_config.model, model_config.download_dir,
                    model_config.load_format, model_config.revision,
                    tp_degree=parallel_config.tp_degree,
                    amp='f32', neuron_config=neuron_config,
                    context_length_estimate=[scheduler_config.max_model_len],
                    n_positions=[scheduler_config.max_model_len],
                    batch_size=scheduler_config.max_num_seqs)
```